packages/psss/src/api/queries/useSingleWeeklyReport.js

```diff
@@ -75,36 +77,52 @@ export const useSingleWeeklyReport = (orgUnit, period, verifiedStatuses, pageQue
         }
 
         const record = results.filter(row => row.period === period);
-        return record.length > 0
-          ? { ...cachedQuery, data: { ...cachedQuery.data, results: record } }
-          : undefined;
+
+        // If there is no results in the cache,
+        // don't try to populate the query with initial data
+        if (record.length === 0) {
+          return undefined;
+        }
+
+        return { ...cachedQuery, data: { ...cachedQuery.data, results: record } };
       },
     },
   );
+};
 
-  if (query.isLoading || query.data.length === 0) {
+export const useSingleWeeklyReport = (orgUnit, period, verifiedStatuses, pageQueryKey) => {
+  const reportQuery = useCachedQuery('weeklyReport', orgUnit, period, pageQueryKey);
+  const confirmedReportQuery = useCachedQuery(
+    'confirmedWeeklyReport',
+    orgUnit,
+    period,
+    pageQueryKey,
+  );
+
+  if (reportQuery.isLoading || confirmedReportQuery.isLoading || reportQuery.data.length === 0) {
     return {
-      ...query,
+      ...reportQuery,
       data: {
         Sites: 0,
         'Sites Reported': 0,
       },
       syndromes: getEmptyTableData(),
       alerts: [],
       unVerifiedAlerts: [],
+      reportStatus: REPORT_STATUSES.OVERDUE,
     };
   }
 
-  const data = query.data[0];
-  const alerts = getAlerts(data);
-
+  const report = reportQuery.data[0];
+  const confirmedReport = confirmedReportQuery?.data[0];
```

Comment:
A syntax I find very readable (although it's 100% fine to keep those as they are!)

```suggestion
  const [report] = reportQuery.data;
  const [confirmedReport] = confirmedReportQuery?.data;
```

Reply:
aah fancy - I like it. I often forget about array de-structuring.